### DISCUSSION TITLE
Failing test at simulator request before test run completion: Simulator shut down prematurely

### DISCUSSION THREAD

**Discussion by jworozco** [cocotb/cocotb#4247] [Q&A]
Hi, 
I created a testbech using pyuvm/cocotb, when i execute the test i get this failure: 
0.00ns INFO     cocotb.regression                  running BaseTest (1/3)
**0.00ns ERROR    cocotb.scheduler                   Failing test at simulator request before test run completion: Simulator shut down prematurely**

i tried expanding the logged info using COCOTB_SCHEDULER_DEBUG:

0.00ns DEBUG    cocotb.scheduler                   Coroutine Scheduler._do_writes yielded <<Event at 0x7f71b5d20580>.wait() at 0x7f71b5d0d380> (mode 1)
0.00ns DEBUG    cocotb.scheduler                   Scheduled coroutine Scheduler._do_writes
0.00ns DEBUG    cocotb.scheduler                   All coroutines scheduled, handing control back to simulator
**0.00ns ERROR    cocotb.scheduler                   Failing test at simulator request before test run completion: Simulator shut down prematurely**
0.00ns DEBUG    cocotb.scheduler                   Issue sim closedown result to regression object
0.00ns DEBUG    cocotb.scheduler                   Unscheduling test <Test BaseTest>

my understanding is the simulator is failing the test, but i dont get any further details on the failure. Has anyone faced that? any way to run it, as to get more info from the simulator as to be able to debug this further?

**Comment by jworozco** [comment]
Never mind, i forgot to setup the clock. With no clock ticking, no events would happen, so simulation does not have anything to do...

**Reply by sconte789** [reply]
I just did the same thing and spend WAAY too long chasing this down. Doh!
